src/main/java/org/tailormap/api/persistence/helper/UploadHelper.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 B3Partners B.V.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+package org.tailormap.api.persistence.helper;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.tailormap.api.util.Constants.UUID_REGEX;
+
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+import org.tailormap.api.controller.UploadsController;
+import org.tailormap.api.repository.UploadRepository;
+
+@Service
+public class UploadHelper {
+
+  private final UploadRepository uploadRepository;
+
+  public UploadHelper(UploadRepository uploadRepository) {
+    this.uploadRepository = uploadRepository;
+  }
+
+  public String getUrlForImage(String imageId, String category) {
+    if (imageId == null || !imageId.matches(UUID_REGEX)) {
+      return null;
+    }
+    return uploadRepository
+        .findByIdAndCategory(UUID.fromString(imageId), category)
```

Comment:
I would prefer not using a regex as these are errorprone and difficult to understand, below relies on builtin behaviour of UUID.

```suggestion
    UUID imageUuid;
    try {
        imageUuid = UUID.fromString(imageId);
    } catch (IllegalArgumentException e) {
        return null;
    }
    return uploadRepository
        .findByIdAndCategory(imageUuid, category)
```